### ISSUE TITLE
Explore Generting or Copying Static Files

### ISSUE THREAD

**Issue by jmbarne3** [UCF/UCF-Announcements-Django#77]
Ensure that the static files are copied into the image.

Definition of Done:

The static files are copied and available within the image.

**Comment by ucfchandra** [comment]
Static files are already copied and available within the image. This is during the `COPY . /app/` step in the Containerfile, as the static folder is copied as well.

**Comment by jmbarne3** [comment]
This is only true if all the assets are already within the static folder when the image is built. The assets under `/static/admin/` will only be there if the local developer creating the image has run `./manage.py collectstatic` at some point. 

If we assume the following scenario, the problem becomes clear:

Developer pulls down the repository for the first time to do some work. They follow the instructions in the readme. Because they have not run `./manage.py collectstatic` yet, when the `/app` folder copy happens, everything in static except for the `/static/admin` directory gets copied. They run the image and the static files for the front end are there and work fine, but the CSS/JS files for the `/admin` area of the site are missing.

The same would be true for if the `/admin` files were there, but the admin app had been updated at some point (this happens occasionally with Django upgrades). While one developer might have the proper static files for the `admin` app getting copied in (because they're run `collectstatic`), another developer might not and could keep overwriting the updates.

In general, the files in the `static` folder should be treated as artifacts. While we track some of the artifacts (the files we generate as part of the project), we don't track all of them (the files that are provided by the django admin app, for example).

TL;DR - we need to work in the `collectstatic` command as part of the build process to make sure all the appropriate files are copied into this directory.

**Comment by ucfchandra** [comment]
Collectstatic is in the build process for the gunicorn and nginx branches. Can we consider this closed?

**Comment by jmbarne3** [comment]
When that code gets reviewed, yes, we can close it. 😄 Issues are issues until they've been resolved in reviewed code. Most organizations keep them open until they're actually merged into master, but I've found it's helpful on longer projects to close them out once they get into the rc branch.

**Comment by ucfchandra** [comment]
Change moved onto RC branch. Closing.
